File: vulnerable_app.js
const express = require("express");
const sqlite3 = require("sqlite3").verbose();
const jwt = require("jsonwebtoken");
const { exec } = require("child_process");

const app = express();
app.use(express.json());

// Hardcoded secret key (Security Issue)
const SECRET_KEY = "mysecretkey123";

// Connect to SQLite database (No input sanitization)
const db = new sqlite3.Database(":memory:");
db.run("CREATE TABLE users (id INTEGER PRIMARY KEY, username TEXT, password TEXT)");

app.post("/login", (req, res) => {
    const { username, password } = req.body;

    // 🚨 SQL Injection Vulnerability 🚨
    const query = `SELECT * FROM users WHERE username = '${username}' AND password = '${password}'`;
    db.get(query, (err, row) => {
        if (row) {
            const token = jwt.sign({ username }, SECRET_KEY);
            res.json({ token });
        } else {
            res.status(401).send("Invalid credentials");
        }
    });
});

// 🚨 Command Injection Vulnerability 🚨
app.get("/exec", (req, res) => {
    const command = req.query.cmd;
    exec(command, (error, stdout) => {
        res.send(stdout || error.message);
    });
});

app.listen(3000, () => {
    console.log("Vulnerable app running on http://localhost:3000");
});

File: VulnerableApp.java
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.*;

import java.sql.*;

@SpringBootApplication
@RestController
public class VulnerableApp {
    // Hardcoded database credentials (Security Issue)
    private static final String DB_URL = "jdbc:sqlite:test.db";
    private static final String DB_USER = "admin";
    private static final String DB_PASS = "password123";

    @GetMapping("/login")
    public String login(@RequestParam String username, @RequestParam String password) {
        try {
            Connection conn = DriverManager.getConnection(DB_URL, DB_USER, DB_PASS);
            Statement stmt = conn.createStatement();

            // 🚨 SQL Injection Vulnerability 🚨
            String query = "SELECT * FROM users WHERE username = '" + username + "' AND password = '" + password + "'";
            ResultSet rs = stmt.executeQuery(query);

            if (rs.next()) {
                return "Welcome, " + username + "!";
            } else {
                return "Invalid credentials!";
            }
        } catch (Exception e) {
            return "Error: " + e.getMessage();
        }
    }

    // 🚨 XSS Vulnerability 🚨
    @PostMapping("/submit")
    public String submit(@RequestBody String userInput) {
        return "User submitted: " + userInput; // No sanitization = XSS
    }

    public static void main(String[] args) {
        SpringApplication.run(VulnerableApp.class, args);
    }
}
